### ISSUE TITLE
Api to reset the state of a scenario in Wiremock?

### ISSUE THREAD

**Issue by ramswaroop** [tomakehurst/wiremock#847]
Let's say a scenario has 3 states and I want to go back to the first state if for some reasons my code (which uses the wiremock's stub) executed 2 states successfully but failed to trigger the 3rd state. 

I want to go back to the first state only after some set specific time. If the 3rd state doesn't get triggered within this set time then go back to state 1.

**Comment by ramswaroop** [comment]
I found a way to reset the states of all scenarios i.e, `WireMock.resetAllScenarios()` but couldn't find a way to reset state of any specific scenario. @tomakehurst by any chance you want to include this feature?

**Comment by tomakehurst** [comment]
I've labelled this a feature request. If you'd like to raise a PR for this, it'll likely happen much sooner.

**Comment by aosorio1** [comment]
I'd love to see this implemented I think it can be used in different uses cases.

**Comment by Nyeng** [comment]
Tried to hack around this feature by setting the last request to   "newScenarioState" : "Started", but this randomly assignes the state to another previous state. If it was possible to set new scenario state back to "started" it would be easy to reset scenarios?

**Comment by CanYumusak** [comment]
@Nyeng https://github.com/tomakehurst/wiremock/pull/1046 should be the Feature Request you are looking for, right?

**Comment by kurimin** [comment]
Is there any news about the PR getting merged? Or is there any workaround at the moment to set the scenario to an specific state?

**Comment by tomakehurst** [comment]
@kurimin the workaround is to create a stub on a URL that doesn't exist in the API you're mocking, and give it a `nextScenarioState` of `STARTED`. Then you just make a request that matches the stub whenever you want to reset the individual scenario.

**Comment by mjustin** [comment]
Does #1721 satisfy the requirements of this request?

**Comment by xqsjdy** [comment]
Hi maintainers,
I'm a computer science undergraduate student (junior year), and I'm
looking to make my first open-source contribution with WireMock.
I'd like to take this issue (#847) about adding a timeout-based reset
for stateful scenarios.
Before starting, I'd like to confirm a few points so my PR matches
the project's expectations:
1. Feature semantics: when a scenario is in an intermediate state and
   no matching stub request arrives within a configurable timeout, the
   scenario should reset to its initial state. Is that the intended
   behavior, and do you have a preferred API for expressing the timeout
   on the scenario/state?
2. Tests: I'll add acceptance tests (real WireMock server + public
   API) covering timeout reset and the non-reset path, following the
   existing test style. Is that the expected scope?
3. Docs: should this be documented in the same PR (wiremock.org
   `_docs`), or is a separate PR preferred?
If this direction looks right, I'd be happy to be assigned. Happy to
start within the next week.
